mungegithub/mungers/issue-creator.go

```diff
@@ -0,0 +1,271 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mungers
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	githubapi "github.com/google/go-github/github"
+	"k8s.io/test-infra/mungegithub/github"
+)
+
+//interface to decouple IssueCreator from github.Config for dependency injection
+type RepoClient interface {
+	GetUser(login string) (*githubapi.User, error)
+	GetLabels() ([]*githubapi.Label, error)
+	ListAllIssues(options *githubapi.IssueListByRepoOptions) ([]*githubapi.Issue, error)
+	NewIssue(title, body string, labels, owners []string) (*github.MungeObject, error)
+
+	getDryRun() bool
+	getOrg() string
+	getProject() string
+
+	RealConfig() *github.Config //Gets the underlying *github.Config or returns nil if the RepoClient is a testing instance
+}
+
+//this type wraps the github.Config type so that it can match the RepoClient interface
+type githubConfig github.Config
+
+//gets the underlying *github.Config
```

Comment:
Throughout: these comments should be complete sentences starting with the function name, to make godoc work.

"RealConfig returns the underlying github.Config"